src/mcelog.c

```diff
@@ -210,95 +213,105 @@ static int socket_reinit(socket_adapter_t *self) {
+                                              mr->uncorrected_err_timed) < 0) {
+    ERROR(MCELOG_PLUGIN ": add corrected timed errors meta data failed");
     plugin_notification_meta_free(n->meta);
     return (-1);
   }
+  if (mr->uncorrected_err_timed_period[0] != '\0')
+    if (plugin_notification_meta_add_string(n, "uncorrected errors time period",
+                                            mr->uncorrected_err_timed_period) <
+        0) {
+      ERROR(MCELOG_PLUGIN ": add corrected errors period meta data failed");
+      plugin_notification_meta_free(n->meta);
+      return (-1);
+    }
 
   return (0);
 }
 
-static int mcelog_submit(mcelog_memory_rec_t mr) {
+static int mcelog_submit(const mcelog_memory_rec_t *mr) {
 
-  value_list_t vl = VALUE_LIST_INIT;
-  vl.values_len = 1;
-  vl.time = cdtime();
+  if (!mr) {
+    ERROR(MCELOG_PLUGIN ": %s: NULL pointer", __FUNCTION__);
+    return -1;
+  }
 
-  sstrncpy(vl.plugin, MCELOG_PLUGIN, sizeof(vl.plugin));
-  sstrncpy(vl.type, "errors", sizeof(vl.type));
-  if (strlen(mr.dimm_name) > 0) {
+  value_list_t vl = {
+      .values_len = 1,
+      .values = &(value_t){.derive = (derive_t)mr->corrected_err_total},
+      .time = cdtime(),
+      .plugin = MCELOG_PLUGIN,
+      .type = "errors",
+      .type_instance = "corrected_memory_errors"};
+
+  if (mr->dimm_name[0] != '\0')
     ssnprintf(vl.plugin_instance, sizeof(vl.plugin_instance), "%s_%s",
-              mr.location, mr.dimm_name);
-  } else
-    sstrncpy(vl.plugin_instance, mr.location, sizeof(vl.plugin_instance));
+              mr->location, mr->dimm_name);
+  else
+    sstrncpy(vl.plugin_instance, mr->location, sizeof(vl.plugin_instance));
 
-  sstrncpy(vl.type_instance, "corrected_memory_errors",
-           sizeof(vl.type_instance));
-  vl.values = &(value_t){.derive = (derive_t)mr.corrected_err_total};
   plugin_dispatch_values(&vl);
 
   ssnprintf(vl.type_instance, sizeof(vl.type_instance),
-            "corrected_memory_errors_in_%s", mr.corrected_err_timed_period);
-  vl.values = &(value_t){.derive = (derive_t)mr.corrected_err_timed};
+            "corrected_memory_errors_in_%s", mr->corrected_err_timed_period);
+  vl.values = &(value_t){.derive = (derive_t)mr->corrected_err_timed};
   plugin_dispatch_values(&vl);
 
   sstrncpy(vl.type_instance, "uncorrected_memory_errors",
            sizeof(vl.type_instance));
-  vl.values = &(value_t){.derive = (derive_t)mr.uncorrected_err_total};
+  vl.values = &(value_t){.derive = (derive_t)mr->uncorrected_err_total};
   plugin_dispatch_values(&vl);
 
   ssnprintf(vl.type_instance, sizeof(vl.type_instance),
-            "uncorrected_memory_errors_in_%s", mr.uncorrected_err_timed_period);
-  vl.values = &(value_t){.derive = (derive_t)mr.uncorrected_err_timed};
+            "uncorrected_memory_errors_in_%s",
+            mr->uncorrected_err_timed_period);
+  vl.values = &(value_t){.derive = (derive_t)mr->uncorrected_err_timed};
   plugin_dispatch_values(&vl);
 
   return 0;
```

Comment:
unrelated to this PR, but I just now noticed this. This file mixes return n and return (n). Could you pick one style?